manticore/manticore.py

```diff
@@ -606,20 +606,25 @@ def run(self, procs=1, timeout=0, should_profile=False):
         :param timeout: Analysis timeout, in seconds
         '''
         assert not self.running, "Manticore is already running."
-        self._start_run()
-
-        self._time_started = time.time()
-        if timeout > 0:
-            t = Timer(timeout, self.terminate)
-            t.start()
         try:
-            self._start_workers(procs, profiling=should_profile)
+            self._start_run()
 
-            self._join_workers()
-        finally:
+            self._time_started = time.time()
             if timeout > 0:
-                t.cancel()
-        self._finish_run(profiling=should_profile)
+                t = Timer(timeout, self.terminate)
+                t.start()
+            try:
+                self._start_workers(procs, profiling=should_profile)
+
+                self._join_workers()
+            finally:
+                if timeout > 0:
+                    t.cancel()
+            self._finish_run(profiling=should_profile)
```

Comment:
Should `_finish_run` be invoked inside the `finally` block?

Reply:
that's not related to this PR but i guess?

i actually do not see the point of the try/finally here, since it doesn't catch anything.